oscore-cf/src/main/java/org/eclipse/californium/oscore/ObjectSecurityLayer.java

```diff
@@ -196,21 +209,26 @@ public void receiveRequest(Exchange exchange, Request request) {
 	}
 
 	@Override
+	//Rewrote this method to always accept unprotected responses
+	//This is needed for error messages
 	public void receiveResponse(Exchange exchange, Response response) {
 		Request request = exchange.getCurrentRequest();
+
+		//Added printing of status information
+		if(!isProtected(response)) {
+			LOGGER.warn("Incoming response is NOT OSCORE protected!");
```

Comment:
Would it be possible to use "warn" only, if the request uses OSCORE?

Reply:
Yes, that makes sense. I can make it so the warning for non-protected replies only triggers if the outgoing request used OSCORE.

Reply:
I restored a method that was in the code previously by the name _responseShouldBeProtected_. It checks if a response is expected to be protected with OSCORE.
I now only warn when the response is expected to be protected. This was added in commit fc51c5a37c249e98107b9c89e4869132ad75fd0a